### PR TITLE
GQL-3: Appending variableKeyMap response to add new JSON fields

### PR DESCRIPTION
**Pull request by eudoroolivares2016**
# Overview

### What is the feature?

Adding JSON fields which were recently indexed into CMR to the JSON key mapping for variables. The purpose of this is to leverage CMR's JSON response's superior speed to ensure that collections which contain a large number of variables do not timeout on API gateway

### What is the Solution?

Added the variables to its `keymap`. Overrode the parent concept class's `parseJson` function to include logic to support consistently returning the metadata in camelCase with the exception of `tags`

List impacted areas.

All JSON responses from any of the concepts which are nested (except tags)

# Testing

### Reproduction steps

Using this query on localhost ensure that even after the changes the response remains the same as the current response from the `umm_json` endpoint

https://studio.apollographql.com/sandbox/explorer?endpoint=http%3A%2F%2Flocalhost%3A3013%2Fdev%2Fapi&explorerURLState=N4IgJg9gxgrgtgUwHYBcQC4QEcYIE4CeABABQAkADgIZ5VwDO6RAajQJZUBGANggJJIKMFAEoiwADpIiRAG7suvEtVoMmlGnXpjJ0mUTZJ6KKkij8kAMwh44VFGwhIp%2BovShtk5gNIICAdxswehd9JHs2WX4wUJkoJ3MKFD4YvRkwBEtDNgcnUIBfKXyQABoQeTwOHgR6DBBdGQkQFS0mpgb9JvizBCSUtqIm5gBGACYABnGAFgB2KYBOKYAOAFoAEQBRAFkAeQB9AAUAJR3mJoKikHygA

### Attachments

Please include relevant screenshots or files that would be helpful in reviewing and verifying this change.

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/cmr/concepts/__tests__/variable.test.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+          const variableConceptId = 'V1200000000-EDSC'
+          const jsonKeys = ['instanceInformation', 'scienceKeywords']
+          const jsonResponse = {
+            data:
+            {
+              hits: 1,
+              took: 9,
+              items: [
+                {
+                  long_name: 'sea surface subskin temperature',
```
Comment (eudoroolivares2016) [inline src/cmr/concepts/__tests__/variable.test.js:24]: I'd prefer to just remove everything here that isn't getting parsed but, I wasn't sure if people thought it helped to have something that looks like a "real" response

File: src/utils/umm/variableKeyMap.json [hunk 2]
```diff
@@ -2,7 +2,12 @@
   "sharedKeys": [
     "conceptId",
     "longName",
-    "name"
+    "name",
```
Comment (macrouch) [inline src/utils/umm/variableKeyMap.json:5]: sort the list

File: src/cmr/concepts/variable.js [hunk 3]
```diff
@@ -27,6 +30,49 @@ export default class Variable extends Concept {
...
+        // Snake case the key requested and any children of that key
+        const cmrKey = snakeCase(jsonKey)
+
+        const { [cmrKey]: keyValue } = normalizedItem
+
+        let formattedKeyValue = keyValue
+
+        // Once accessed, ensure that the child key/value pairs are consistently returning in `camelCase`
+        // Note applying camelcase logic to all keys in `concepts` will not work because of `tags` business logic
+        if (jsonKey === 'scienceKeywords' || jsonKey === 'instanceInformation') {
```
Comment (macrouch) [inline src/cmr/concepts/variable.js:67]: If the only field we know of that we don't want camelCased is 'tags', then use `!== 'tags'`, that will ensure future fields that need to follow the naming scheme will match without needing more conditionals.

That logic also seems like it should be uniform across all concepts. Is this line the only thing different between concept.js and variable.js? If both those things are true this logic should be in concept.js

Comment (eudoroolivares2016) [reply src/cmr/concepts/variable.js:67]: Yea I did it that way initially. It is also going to be fields within the `data` inside of the association-details which work the same way. Yes that is the only difference it is simple to add this to concepts instead

File: src/cmr/concepts/__tests__/concept.test.js [hunk 4]
```diff
@@ -47,5 +47,123 @@ describe('collection', () => {
...
+              s3CredentialsApiDocumentationUrl: 'https://s3Docsmock.html',
+              s3CredentialsApiEndpoint: 'https://s3mock.html'
+            },
+            url: 'https://mock.html'
+          })
+        })
+      })
+
+      describe('the nested key is tag', () => {
+        test('the child-keys of the tag return snake_cased', () => {
```
Comment (macrouch) [inline src/cmr/concepts/__tests__/concept.test.js:128]: Change this test name to be more clear about what is happening in the code. The child keys aren't being returned snake_cased, they are being returned in their original casing

**Review by macrouch — APPROVED** [review]

**Review by dpesall — CHANGES_REQUESTED** [review]
We should investigate that CMR’s bulk_indexing for variables is working as expected before merging. I’ve found some new info that suggests simply the bulk_index for all providers may not be executing for ALL providers

**Comment by eudoroolivares2016** [comment]
I closed this pull request because until we resolve issues around bulk-reindexing we cannot reliably ensure that we retrieve these fields from the `.json` endpoint and that they match the `.umm_json` endpoint for existing variables as both systems propagate through the envs. A ticket was written to address the bulk_reindexing issue on CMR

**Pull request closed without merging**